### PR TITLE
hiawatha: GitHub -> GitLab, clean-up

### PR DESCRIPTION
**Pull request by Anton-Latukha**
Developer deleted account on GitHub. As a GitLab proof - official page:
https://www.hiawatha-webserver.org/download
points to GitLab.

* Web server officially created for `mbedtls`, so use bundled
* Assert not needed
* Build with cmake/ninja
* Removed hack, no longer needed

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Anton-Latukha** [comment]
Calling package maintainer: @ndowens

File: pkgs/servers/http/hiawatha/default.nix [hunk 1]
```diff
@@ -1,50 +1,50 @@
...
-    sha256 = "1f2hlw2lp98b4dx87i7pz7h66vsy2g22b5adfrlij3kj0vfv61w8";
+  src = fetchFromGitLab {
+    owner = "hsleisink";
+    repo = "hiawatha";
+    rev = "v${version}";
+    sha256 = "1428byx0xpzzwyc0j157q70sjx18dykvg6fd5vp70kj85ank0xpa";
   };
 
-  buildInputs =  [ cmake libxslt zlib libxml2 ] ++ stdenv.lib.optional enableSSL openssl ;
+  buildInputs =  [ cmake ninja libxslt libxml2 ];
```
Comment (jtojnar) [inline pkgs/servers/http/hiawatha/default.nix]: At least `cmake` and `ninja` should go to `nativeBuildInputs`.

**Comment by jtojnar** [comment]
We should probably add those flags and build this with `mbedtls`.

**Comment by Anton-Latukha** [comment]
Do make this flags:
```
"The mbed TLS library must be compiled with MBEDTLS_THREADING_PTHREAD and MBEDTLS_THREADING_C enabled."
```
default in `mbedtls`?

Seems like a reasonable thing.

`mbedtls` also has test suite written in Perl. Should I enable it. If so, launch tests as postinstall, or make it into a test.

**Comment by jtojnar** [comment]
Sounds good to me.

Enabling test suite is also always good if it is not too much hassle.

**Pull request merged**